core/src/gl/error.h

```diff
@@ -18,10 +18,18 @@ class Error {
      */
     static bool hadGlError(const std::string& _locationTag);
 
+    static void glError(const char* stmt, const char* fname, int line);
+
 private:
 
     static std::unordered_map<GLenum, std::string> s_GlErrorCodesToStrings;
 
 };
 
+#ifdef TANGRAM_DEBUG
+#define GL_CHECK(STMT) do { STMT; Tangram::Error::glError(#STMT, __FILE__, __LINE__); } while (0)
```

Comment:
Yeah I'm cool with that.
On May 4, 2016 1:00 AM, "Karim Naaji" <notifications@github.com> wrote:

> In core/src/gl/error.h
> <https://github.com/tangrams/tangram-es/pull/712#discussion_r61907356>:
>
> >  private:
> >
> >      static std::unordered_map<GLenum, std::string> s_GlErrorCodesToStrings;
> >
> >  };
> >
> > +#ifdef TANGRAM_DEBUG
> > +#define GL_CHECK(STMT) do { STMT; Tangram::Error::glError(#STMT, __FILE__, __LINE__); } while (0)
>
> We don't have to wrap STMT in a set of parenthesis to be able to write
> GL_CHECK(), the macro would be an extra ; in that case.
> I'm not sure we want something that clever, there's a pretty light set of
> gl functions we use that return a value, we already have all of the
> information we need to debug with this.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/tangrams/tangram-es/pull/712/files/86ce6589acb568c5109f181528406482e85286ed#r61907356>
>